service/db/dgclient.go

```diff
@@ -341,6 +260,21 @@ func (s DGClient) GetAllByClusterAndType(meta string, cluster string) (map[strin
 	return m, nil
 }
 
+//GetSchema - get all predicates
+func (s DGClient) GetSchema() ([]*api.SchemaNode, error) {
+	q := `
+		schema {}
+	`
+	resp, err := s.dc.NewTxn().Query(context.Background(), q)
```

Comment:
would it make sense to try to encapsulate actual DB interactions to one place like you may be trying to do with `GetQueryResult` method above?

essentially, could `GetSchema` use `GetQueryResult` so that it doesn't also need the code repeated to do a DB query?

Reply:
For my understanding, the dgclient.go is sort of one place for actual DB interactions. We better separate those functions in case of other functions can call them individually. Does this make sense to you? What do you think?